File: ios-native/Nucleo/ActionMap.swift
import Foundation
import SwiftData

struct ActionMap: Codable, Identifiable, Sendable {
    var id: String
    var title: String
    var category: String?
    var pinnedAt: Date?
    var sourceType: String
    var document: Data
    var currentStep: Int
    var isComplete: Bool
    var viewAll: Bool
    var createdAt: Date
    var updatedAt: Date
}

@Model
final class CachedMap {
    @Attribute(.unique) var id: String
    var payload: Data
    var updatedAt: Date

    init(map: ActionMap) throws {
        id = map.id
        payload = try JSONEncoder().encode(map)
        updatedAt = map.updatedAt
    }

    func decode() throws -> ActionMap { try JSONDecoder().decode(ActionMap.self, from: payload) }
}

File: android-native/app/src/main/java/com/nucleo/app/ActionMap.kt
package com.nucleo.app

import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Serializable
data class ActionMap(
    val id: String,
    val title: String,
    val category: String? = null,
    val pinnedAt: Long? = null,
    val sourceType: String,
    val document: String,
    val currentStep: Int = 0,
    val isComplete: Boolean = false,
    val viewAll: Boolean = false,
    val createdAt: Long,
    val updatedAt: Long
)

@Entity(tableName = "maps")
data class CachedMap(@PrimaryKey val id: String, val payload: String, val updatedAt: Long)
